lib/hcrypto/ui.c

```diff
@@ -125,7 +125,7 @@ read_string(const char *preprompt, const char *prompt,
     sigemptyset(&sa.sa_mask);
     sa.sa_flags = 0;
     for(i = 1; i < sizeof(sigs) / sizeof(sigs[0]); i++)
-	if (i != SIGALRM)
+	if (i != SIGALRM && i != SIGWINCH)
```

Comment:
SIGWINCH isn't defined on all platforms so its use must be conditionalized 

   #if defined(SIGWINCH)
       && i != SIGWINCH
   #endif

Reply:
How does this look:
`    if (i != SIGALRM
#if defined(SIGWINCH)
            && i != SIGWINCH
#endif
       )
`

Reply:
Added the whole line to the condition, and tried to match the indentation style.

Reply:
This function makes me sad.  Its purpose is to prompt, of course, possibly with echo off, and to terminate when the user... types `^C` I guess.  But messing with signal handling in a library function is a bad idea.  A better way to write this would be to not use stdio, block signals, put the terminal into raw mode, and look for `^C` (or whatever it's been mapped to with `stty`) instead of catching signals, then restore the tty mode to cooked (if it had been cooked) or whatever it was set to prior to returning.

Anyways, the change looks fine to me.  Can you squash commits and give the one commit a subject like "lib/hcrypto/ui: ignore SIGWINCH" or similar?  Or would you like us to do this?